vision/use-cases/video-thumbnail-generation/video_thumbnail_generation.ipynb

```diff
@@ -0,0 +1,804 @@
+        "Your task is to find the best 4 thumbnails from a given video along with short, and meaningful captions that is good for marketing.\n",
+        "Consider the following rules while generating thubmnails:\n",
+        "\n",
+        "- Thumbnail should have clear focus on the key objects and people, less focus on background\n",
+        "- Thumbnail image should be high quality and bright, avoid blurry images\n",
+        "- Thumbnail image and caption together tell a story\n",
+        "- Thumbnail caption is good for marketing\n",
+        "\n",
+        "Provide response in a proper JSON format similar to following:\n",
+        "{\n",
+        "    \\\"timestamp\\\": [],\n",
+        "    \\\"caption\\\": []\n",
+        "}\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1e718c0ce536"
+      },
+      "source": [
+        "### calling Gemini API"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 54,
+      "metadata": {
+        "id": "c653891025b2"
+      },
+      "outputs": [],
+      "source": [
+        "response_text2 = call_gemini_1_5_Pro(\n",
+        "    prompt=advanced_prompt,\n",
+        "    gcs_video_path=video_uri,\n",
+        "    model=model,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ea0736fc652d"
+      },
+      "source": [
+        "### showing JSON output string"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 55,
+      "metadata": {
+        "id": "8bf044c5f747"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "```json\n",
+            "{\n",
+            "    \"timestamp\": [\n",
+            "        \"0:05\",\n",
+            "        \"0:17\",\n",
+            "        \"0:32\",\n",
+            "        \"1:12\"\n",
+            "    ],\n",
+            "    \"caption\": [\n",
+            "        \"Last minute packing woes? Download Google Trips before you go!\",\n",
+            "        \"Lost without data? Google Trips works offline to keep your travel plans on track.\",\n",
+            "        \"Ditch the tourist traps! Curate your dream itinerary with Google Trips.\",\n",
+            "        \"See more, plan less. Download Google Trips for stress-free travel.\"\n",
+            "    ]\n",
+            "}\n",
+            "```\n",
+            "\n"
+          ]
+        }
+      ],
+      "source": [
+        "print(response_text2)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1f7e4a18918b"
+      },
+      "source": [
+        "### displaying final thumbnails with captions"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 56,
+      "metadata": {
+        "id": "5558f8fd147a"
+      },
+      "outputs": [
+        {
+          "data": {
```

Comment:
Remove this image output. It makes the notebooks too large

Reply:
Done